### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by kaitcham**
For this branch, I followed these [requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md)

**Required Changes**
- A method which implements the has-many side of the relationship between the Person and Rental

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    super()
+    @label = label
+    @students = []
+  end
+
+  def add_student(student)
+    @students.push(student)
+    student.classroom = self
+  end
+end
```
Comment (t-yanick) [inline classroom.rb:15]: - [ ] To implement the `belong-to` side of the relationship between `Student` and `Classroom`, a method that serves as the 'belongs-to' side (a student belongs to a classroom) should be implemented in the `Student` class. Checkout this [link](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships`)  for clarifications. Something like so: 
```ruby
  def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  attr_accessor :date
+  attr_reader :book, :person
+
+  def initialize(date, book, person)
+    super()
+    @date = date
+    @book = book
+    book.rental.push(self) unless book.rental.includes?(self)
+    @person = person
+    person.rental.push(self) unless person.rental.includes?(self)
```
Comment (t-yanick) [inline rental.rb:11]: - [ ] To implement the `has-many` side of the relationship between `Person` and `Rental`, a method that serves as the 'has-many' side (a person has many rentals) should be implemented in the `Person` class. Checkout this [link](`https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship`) for clarifications. Something like so: 
```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
+  attr_reader :book, :person
```
Comment (t-yanick) [inline rental.rb:3]: - [ ] Please kindly set `setters` for `@book and @person` instance variables, are they are modifies within the `Rental` class. Something like so: 
```
attr_accessor  :book, :person
```

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @kaitcham, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 3 above)
Comment (sja-thedude) [reply rental.rb]: Implemented 👍

File: classroom.rb (hunk 1 above)
Comment (sja-thedude) [reply classroom.rb:17]: Implemented 👍

**Review by Meltrust — APPROVED** [review]
Hi @kaitcham,


## Wow, you did it 🎉
Brilliant
## Thank you for implementing the changes requested by the previous reviewer 💪 🥇 ㊗️ 
### Well done!




Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### To highlight

- Thank you for updating your person class, as requested✔️ 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**